GW2EIEvtcParser/EIData/ProfHelpers/Necromancer/ScourgeHelper.cs

```diff
@@ -21,6 +24,17 @@ internal static class ScourgeHelper
             new BuffGainCastFinder(SandstormShroudSkill, DesertShroudBuff).UsingDurationChecker(3500),
             // new EXTBarrierCastFinder(DesertShroud, DesertShroud),
             new EXTBarrierCastFinder(SandCascadeSkill, SandCascadeBarrier),
+            new BuffGainCastFinder(SadisticSearing, SadisticSearing),
+            new BuffLossCastFinder(SadisticSearingActivation, SadisticSearing).UsingChecker((blcf, combatData, agentData, skillData) => 
+            {
+                AbstractBuffEvent abe = combatData.GetBuffData(SadisticSearing).Where(buffEvent => buffEvent is BuffApplyEvent).LastOrDefault(x => x.Time < blcf.Time);
+                AbstractHealthDamageEvent ahde = combatData.GetDamageData(ManifestSandShadeShadeHit).LastOrDefault(x => x.Time < blcf.Time);
+                if (abe.Time < ahde.Time && ahde.Time < blcf.Time)
```

Comment:
You should null check both abe and ahde.

What is the logic behind the presence of abe? Wouldn't doing ahde = ...LastOrDefault(x => abs(x.Time - blcf.Time) < ServerDelayConstant) and then return ahde != null be enough? Abe could be null if the player starts the log with the buff and buff initials are missing for some reason.

Reply:
scratch the abs(...) should be blcf.Time - x.Time < ServerDelayConstant to make sure events are properly ordered.